src/irmin/layer.mli

```diff
@@ -0,0 +1,58 @@
+(*
+ * Copyright (c) 2013-2019 Thomas Gazagnaire <thomas@gazagnaire.org>
+ *                         Ioana Cristescu <ioana@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+(** Signature for layer stores. *)
+module type STORE = sig
+  include S.STORE
+
+  val freeze :
+    ?min:commit list -> ?max:commit list -> ?squash:bool -> repo -> unit Lwt.t
+  (** [freeze min max squash t] copies the upper layer of [t] into the lower
+      layer and clears the upper.
+
+      If [squash] then copy the commits, nodes and contents of the [max]
+      commits. Otherwise, copy the closure of commits, nodes and contents
+      between the commits [min] and [max]. *)
```

Comment:
What happens if `max` is not set? What happens is `max` is not set AND `squash` is set? In these cases, would it make sense to look at the existing branches' HEAD in upper?

Reply:
The [current heads are used](https://github.com/mirage/irmin/pull/882/files#diff-e1c2498849d6289c1a35dec9cab695edR363) when max is empty. I added a line in the doc about it.